sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1267,6 +1394,8 @@ class HistGradientBoostingRegressor(RegressorMixin, BaseHistGradientBoosting):
           features.
         - str array-like: names of categorical features (assuming the training
           data has feature names).
+        - `"by_dtype"`: Pandas categorical dtypes are considered categorical.
+          The input must be a pandas DataFrame to use this feature.
```

Comment:
I guess and hope that we can extend this in the future and also treat string columns of any dataframe the same way.

Reply:
String columns might contain free text (e.g. a title, a description, people full names,...) which are not really categorical features directly. They would require some kind of text tokenization preprocessing first.

I have the feeling that it's better to error on those and let the user either type those columns excplicitly as categorical or do some appropriate preprocessing.